### PR TITLE
Benchmarks: Fix Bug - Fix bug when validate the raw data format.

### PR DESCRIPTION
**Pull request by guoshzhao**
Change the validation of raw data from:
```
    instance of List[List[Number]] for BenchmarkType.MODEL, and BenchmarkType.DOCKER.
    instance of List[str] for BenchmarkType.MICRO.
```
To
```
    instance of List[List[Number]] for BenchmarkType.MODEL.
    instance of List[str] for BenchmarkType.DOCKER.
    instance of List[List[Number]] or List[str] for BenchmarkType.MICRO.
```

### CODE REVIEW

File: superbench/benchmarks/base.py [hunk 1]
```diff
@@ -191,30 +191,36 @@ def __check_raw_data(self):
...
-                        if not isinstance(run, list):
-                            is_valid = False
-                            break
-                        for value in run:
-                            if not isinstance(value, numbers.Number):
-                                is_valid = False
-                                break
-                    elif self._benchmark_type in [BenchmarkType.MICRO]:
-                        is_valid = isinstance(run, str)
+        def is_list_str(data):
```
Comment (cp5555) [inline superbench/benchmarks/base.py:198]: __is_list_str()

File: superbench/benchmarks/base.py [hunk 2]
```diff
@@ -191,30 +191,36 @@ def __check_raw_data(self):
...
+            if isinstance(data, list) and all(isinstance(item, str) for item in data):
+                return True
+            return False
 
+        def is_list_list_number(data):
```
Comment (cp5555) [inline superbench/benchmarks/base.py:203]: __is_list_list_number()

File: superbench/benchmarks/base.py (hunk 1 above)
Comment (guoshzhao) [reply superbench/benchmarks/base.py:198]: Cool.

File: superbench/benchmarks/base.py (hunk 2 above)
Comment (guoshzhao) [reply superbench/benchmarks/base.py:203]: OK.

**Review by cp5555 — APPROVED** [review]

**Review by cp5555 — APPROVED** [review]

File: superbench/benchmarks/base.py [hunk 3]
```diff
@@ -168,18 +168,16 @@ def __check_summarized_result(self):
...
         """
-        for metric in self._result.result:
-            is_valid = isinstance(self._result.result[metric], list)
-            if is_valid:
-                for value in self._result.result[metric]:
-                    if not isinstance(value, numbers.Number):
-                        is_valid = False
-                        break
+        def __is_list_number(data):
+            if isinstance(data, list) and all(isinstance(item, numbers.Number) for item in data):
```
Comment (abuccts) [inline superbench/benchmarks/base.py:172]: merge to `__is_list_type(data, t)`?

**Review by abuccts — APPROVED** [review]

File: superbench/benchmarks/base.py (hunk 3 above)
Comment (guoshzhao) [reply superbench/benchmarks/base.py]: Done.

File: superbench/benchmarks/base.py [hunk 4]
```diff
@@ -162,24 +162,30 @@ def __check_result_type(self):
 
         return True
 
+    def __is_list_type(self, data, type):
```
Comment (abuccts) [inline superbench/benchmarks/base.py:165]: `type` is builtin function so may be not a good var name

File: superbench/benchmarks/base.py [hunk 5]
```diff
@@ -162,24 +162,30 @@ def __check_result_type(self):
...
+        if isinstance(data, list) and all(isinstance(item, type) for item in data):
+            return True
+        return False
+
+    def __is_list_list_type(self, data, type):
+        if (
+            isinstance(data, list) and all(isinstance(item, list) for item in data)
```
Comment (abuccts) [inline superbench/benchmarks/base.py:172]: `__is_list_type(data, list)`?

File: superbench/benchmarks/base.py [hunk 6]
```diff
@@ -191,30 +197,29 @@ def __check_raw_data(self):
...
-                        for value in run:
-                            if not isinstance(value, numbers.Number):
-                                is_valid = False
-                                break
-                    elif self._benchmark_type in [BenchmarkType.MICRO]:
-                        is_valid = isinstance(run, str)
+        def __is_list_str(data):
+            if isinstance(data, list) and all(isinstance(item, str) for item in data):
+                return True
+            return False
```
Comment (abuccts) [inline superbench/benchmarks/base.py:207]: remove?

Comment (guoshzhao) [reply superbench/benchmarks/base.py:207]: Done.

File: superbench/benchmarks/base.py (hunk 4 above)
Comment (guoshzhao) [reply superbench/benchmarks/base.py:165]: OK

File: superbench/benchmarks/base.py (hunk 5 above)
Comment (guoshzhao) [reply superbench/benchmarks/base.py:172]: Done.

**Review by cp5555 — APPROVED** [review]

**Pull request merged**